src/client/theme-default/components/AlgoliaSearchBox.vue

```diff
@@ -54,11 +58,21 @@ function update(options: any) {
 }
 
 function initialize(userOptions: any) {
+  // if the user has multiple locales, the search results should be filtered
+  // based on the language
+  const facetFilters = props.multilang ? ['language:' + siteData.value.lang] : []
+
   docsearch(
     Object.assign({}, userOptions, {
       container: '#docsearch',
 
-      searchParameters: Object.assign({}, userOptions.searchParameters),
+      searchParameters: Object.assign({}, userOptions.searchParameters, {
+        // pass a custom lang facetFilter to allow multiple language search
+        // https://github.com/algolia/docsearch-configs/pull/3942
+        facetFilters: facetFilters.concat(
+          (userOptions.searchParameters || {}).facetFilters || []
```

Comment:
What about `userOptions.searchParameters?.facetFilters || []` ?

Reply:
I would have used it but the syntax is not transpiled and has very recent support only

Reply:
Not transpiled? I thought it would be 🤔

Reply:
oh, it's actually transpiled when building with vitepress! Not when building vitepress itself. It should be fine then